islandora.module

```diff
@@ -466,27 +466,54 @@ function islandora_entity_view(array &$build, EntityInterface $entity, EntityVie
   // Ensure the entity matches the route.
   if ($entity === $route_match_item) {
     if ($display->getComponent('field_gemini_uri')) {
-      $gemini = \Drupal::service('islandora.gemini.lookup');
-      if ($gemini instanceof GeminiLookup) {
-        $fedora_uri = $gemini->lookup($entity);
-        if (!is_null($fedora_uri)) {
-          $build['field_gemini_uri'] = [
-            '#type' => 'container',
-            '#attributes' => [
-              'id' => 'field-gemini-uri',
-            ],
-            'internal_label' => [
-              '#type' => 'item',
-              '#title' => t('Fedora URI'),
-              'internal_uri' => [
-                '#type' => 'link',
-                '#title' => t("@url", ['@url' => $fedora_uri]),
-                '#url' => Url::fromUri($fedora_uri),
-              ],
-            ],
-          ];
+      $mapper = \Drupal::service('islandora.entity_mapper');
+      $flysystem_config = Settings::get('flysystem');
+      $fedora_root = $flysystem_config['fedora']['config']['root'];
+      $fedora_root = rtrim($fedora_root, '/');
+
+      if ($entity->getEntityTypeId() == 'media') {
+        // Check if the source file is in Fedora or not.
+        $media_source_service = \Drupal::service('islandora.media_source_service');
+        $source_file = $media_source_service->getSourceFile($entity);
+        $uri = $source_file->getFileUri();
+        $scheme = \Drupal::service('file_system')->uriScheme($uri);
```

Comment:
This is deprecated in D9:
Replaced with `\Drupal\Core\StreamWrapper\StreamWrapperManagerInterface::getScheme()`